pkg/runctl/logging.go

```diff
@@ -47,5 +48,32 @@ func getPipelineRunInfoForLogging(run *v1alpha1.PipelineRun) []interface{} {
 			"pipelineRunExecAuxNamespace", run.Status.AuxiliaryNamespace,
 		)
 	}
+	for loggingLabel, accessor := range customLoggingDetails {
+		value := getValueByAccessor(run, accessor)
+		if value != "" {
+			kvs = append(kvs, loggingLabel, value)
+		}
+	}
+
 	return kvs
 }
+
+func getValueByAccessor(run *v1alpha1.PipelineRun, accessor cfg.PipelineRunAccessor) string {
+	switch accessor.Kind {
+	case cfg.KindLabelAccessor:
+		labels := run.GetLabels()
+		if labels == nil || accessor.Name == "" {
+			return ""
+		}
+		return labels[accessor.Name]
+
+	case cfg.KindAnnotationAccessor:
+		annotations := run.GetAnnotations()
+		if annotations == nil || accessor.Name == "" {
+			return ""
+		}
+		return annotations[accessor.Name]
+	default:
+		return ""
+	}
```

Comment:
It's ugly to have this logic here. Move this into a method of LabelAccessor. It would be even better to have different types for each kind of accessor.

Reply:
Done: https://github.com/SAP/stewardci-core/pull/412/commits/6b7c08f4efca6fab957ec3c1ea8cc98c6b9ed1f8